Mathlib/RingTheory/Valuation/Extension.lean

```diff
@@ -160,6 +158,58 @@ theorem instIsLocalHomValuationInteger {S ΓS : Type*} [CommRing S]
 
 end integer
 
+section AlgebraInstances
+
+open IsLocalRing Valuation ValuationSubring
+
+variable {K L Γ₀ Γ₁ : outParam Type*} [Field K] [Field L] [Algebra K L]
+  [LinearOrderedCommGroupWithZero Γ₀] [LinearOrderedCommGroupWithZero Γ₁] (vK : Valuation K Γ₀)
+   (vL : Valuation L Γ₁) [vK.HasExtension vL]
+
+local notation "K₀" => Valuation.valuationSubring vK
+local notation "L₀" => Valuation.valuationSubring vL
+
+lemma algebraMap_mem_valuationSubring (x : K₀) : algebraMap K L x ∈ L₀ := by
+  rw [mem_valuationSubring_iff, ← _root_.map_one vL, ← _root_.map_one (algebraMap K L),
+    val_map_le_iff (vR := vK), _root_.map_one]
+  exact x.2
+
+instance instAlgebra_valuationSubring : Algebra K₀ L₀ :=
+  inferInstanceAs (Algebra vK.integer vL.integer)
+
+@[simp]
+lemma coe_algebraMap_valuationSubring_eq (x : K₀) :
+  (algebraMap K₀ L₀ x : L) = algebraMap K L (x : K) := rfl
+
+instance instIsScalarTower_valuationSubring : IsScalarTower K₀ K L :=
+  inferInstanceAs (IsScalarTower vK.integer K L)
+
+instance instIsScalarTower_valuationSubring' : IsScalarTower K₀ L₀ L :=
+  instIsScalarTowerInteger
+
+instance : IsLocalHom (algebraMap K₀ L₀) := instIsLocalHomValuationInteger
+
+lemma algebraMap_mem_maximalIdeal_iff {x : K₀} :
+    algebraMap K₀ L₀ x ∈ (maximalIdeal L₀) ↔ x ∈ maximalIdeal K₀ := by
+  simp only [mem_maximalIdeal_iff, coe_algebraMap_valuationSubring_eq,
+    val_map_lt_one_iff vK vL]
+
+lemma maximalIdeal_comap_algebraMap_eq_maximalIdeal :
+    (maximalIdeal L₀).comap (algebraMap K₀ L₀) = maximalIdeal K₀ :=
+  Ideal.ext fun _ ↦ by rw [Ideal.mem_comap, algebraMap_mem_maximalIdeal_iff]
+
+instance : Ideal.LiesOver (maximalIdeal L₀) (maximalIdeal K₀) :=
```

Comment:
Can you give it a name? The autogenerated one is `Valuation.HasExtension.instAlgebraResidueFieldSubtypeMemValuationSubringValuationSubring`...

More importantly, I have the impression this create a diamond: renaming it to `foo` the following fails
```lean
example {K Γ₀ : Type*} [Field K] [LinearOrderedCommGroupWithZero Γ₀] (vK : Valuation K Γ₀)
    [vK.HasExtension vK] : foo vK vK = Algebra.id _ := by
  with_reducible_and_instances
  rfl
```
But I think the problem is with `IsLocalRing.ResidueField.instAlgebra` and should probably be solved in another PR.

Reply:
I renamed it to `instAlgebra_residueField`. 

Indeed, the problem seems to be with `IsLocalRing.ResidueField.instAlgebra`, since the following also fails:
```lean
example {R : Type*} [CommRing R] [IsLocalRing R] [IsLocalHom (algebraMap R R)] : 
  IsLocalRing.ResidueField.instAlgebra (R := R) (S := R) = Algebra.id _ := by
  with_reducible_and_instances
  rfl
```

Reply:
That's a good point; I deleted it.